### PR TITLE
Fix for #818 - Added support for quoted values

### LINKED ISSUE #818: Can't set annotation with colons
```
$ kustomize edit add annotation -- project/url:http://my_awesome_project
Error: invalid annotation: project/url:http://my_awesome_project (too many colons)
...
```

```
$ kustomize version 
Version: {KustomizeVersion:2.0.1 GitCommit:ce7e5ee2c30cc5856fea01fe423cf167f2a2d0c3 BuildDate:2019-02-08T00:57:04Z GoOs:unknown GoArch:unknown}
```

:-(

### PR DESCRIPTION
**Pull request by narayanan**
Fixes #818

### CODE REVIEW

**Comment by narayanan** [comment]
Singed CLA

File: pkg/commands/edit/add/addmetadata.go [hunk 1]
```diff
@@ -135,13 +135,28 @@ func (o *addMetadataOptions) convertToMap(arg string) (map[string]string, error)
...
 		} else {
 			result[kv[0]] = ""
 		}
+
+		// remove quotes if value is quoted
+		if len(result[kv[0]]) > 0 &&
+			result[kv[0]][:1] == "\"" &&
+			result[kv[0]][len(result[kv[0]])-1:] == "\"" {
+			result[kv[0]] = result[kv[0]][1 : len(result[kv[0]])-1]
+		}
```
Comment (monopole) [inline pkg/commands/edit/add/addmetadata.go]: Thanks so much for writing tests!

Suggest that you simplify this code by taking the form:
```
c := strings.Index(input, ":")
if c < 0 {
  return nil, fmt.Error("need k:v pair where v may be quoted, but got '%s'", input)
}
k := input[:c]
v := trimQuotes(input[c+1:])
...
```
where trimQuotes is as done at https://stackoverflow.com/a/44222891

Comment (narayanan) [reply pkg/commands/edit/add/addmetadata.go]: Done...

Comment (narayanan) [reply pkg/commands/edit/add/addmetadata.go]: Simplified the logic & moved the trim quotes logic to separate function.

**Pull request merged**